backend_tv/lifter.cpp

```diff
@@ -1391,22 +1401,96 @@ class arm2llvm {
     return CurInst->getOperand(idx).getImm();
   }
 
+  enum ExtendType { UXTB, UXTH, UXTW, UXTX, SXTB, SXTH, SXTW, SXTX };
+
+  // Follows the "Library pseudocode for aarch64/instrs/extendreg/ExtendReg"
+  // from ARM manual
+  // val is always 64 bits and shiftAmt is always 0-4
+  Value *extendAndShiftValue(Value *val, enum ExtendType extType,
+                             int shiftAmt) {
+    assert(val->getType()->getIntegerBitWidth() == 64);
+    assert(shiftAmt >= 0 && shiftAmt <= 4);
+
+    // size is always 64 for offset shifting instructions
+    //    auto size = getInstSize(opcode);
+    auto size = 64;
+    auto ty = getIntTy(size);
+    auto isSigned = extType / 4;
```

Comment:
I think bit manipulation would be more clear than division here, I'd write this as `(extType & 0x4) != 0`